fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -0,0 +1,197 @@
+                                   gridspec_kw={
+                                       'width_ratios': [10, 20, 7.8],
+                                       'wspace': 0.01})
+            legend_ax = 2
+
+        # Plot linear performance profile
+        create_plot(ax[0], step_values, acc.keys())
+        ax[0].set_xlim(1, linear_upper_limit)
+        ax[0].set_xticks([1, 2, 4, 6, 8, 10])
+        ax[0].set_xticklabels(['$1$', '$2$', '$4$', '$6$', '$8$', '$10$'])
+        ax[0].yaxis.set_ticks_position('left')
+
+        if use_log_plot:
+            # Plot log performance profile
+            create_plot(ax[1], step_values, acc.keys())
+            ax[1].set_xlim(
+                linear_upper_limit,
+                min(max_value+1, 10000))
+            ax[1].set_xscale('log')
+            ax[1].set_xticks([100, 1000, 10000])
+            ax[1].set_xticklabels(['$10^2$', '$10^3$', '$10^4$'])
+            ax[1].yaxis.set_ticks_position('right')
+            ax[1].tick_params(axis='y', labelcolor='white')
+
+        # legend
+        ax[legend_ax].axis('off')
+        handles, labels = ax[0].get_legend_handles_labels()
+        wrapped_labels = ['\n'.join(wrap(l, 22)) for l in labels]
+        ax[legend_ax].legend(handles, wrapped_labels, loc=2, prop={'size': 7})
+
+        # Common parts
+        plt.ylim(0.0, 1.0)
+        fig.suptitle("Performance profile - {}".format(name))
+
+        # add a big axis, hide frame
+        fig.add_subplot(111, frameon=False)
+        # hide tick and tick label of the big axis
+        plt.tick_params(labelcolor='none',
+                        top=False,
+                        bottom=False,
+                        left=False,
+                        right=False)
+        # the xlabel has added white space to (empirically) make it look
+        # centred under the plots (and not the legend)
+        plt.xlabel("f                       ")
+        plt.ylabel("fraction for which solver within f of best")
+        ax[0].set_ylim(0.0, 1.05)
+
+        plt.savefig(this_filename, dpi=150)
+
+    return figure_path
+
+def create_plot(ax, step_values, keys):
+    """
+    Function to draw the profile on a matplotlib axis
+
+    :param ax : A matplotlib axis to be filled
+    :type ax :  an `.axes.SubplotBase` subclass of `~.axes.Axes`
+                (or a subclass of `~.axes.Axes`)
+    :param step_values : a sorted list of the values of the metric
+                         being profiled
+    :type step_values : list of float
+    """
+
+    lines = ["-", "-.", "--", ":"]
+    # use only 9 of matplotlib's colours, as this will give us
+    # 9 * 4 = 36 line/colour combinations, as opposed to
+    # 10 * 4 / 2 = 20 if we used all 10
+    colors = ["C0","C1","C2","C3","C4","C5","C6","C7","C8" ]
+
+    no_failures = np.zeros(len(step_values), dtype=np.int8)
+    huge = 1.0e20  # set a large value as a proxy for infinity
+    for i, solver_values in enumerate(step_values):
```

Comment:
`no_failures` and `labels` could be removed if this was done in 1 loop rather than three with:
`for i, (solver, solver_values) in enumerate(keys, step_values):`

Reply:
Good point -- fixed.